src/test/php/org/bovigo/vfs/vfsStreamWrapperFileTestCase.php

```diff
@@ -454,4 +454,22 @@ public function cannotReadFileFromNonReadableDir()
         $this->bar->chmod(0);
         $this->assertFalse(@file_get_contents($this->baz1URL));
     }
+
+    /**
+     * @test
+     * @see   https://github.com/mikey179/vfsStream/issues/155
+     * @since 1.6.5
+     */
+    public function errorOnOpen()
+    {
+        $file = $this->baz1;
+        $path = $file->url();
+        $this->assertNotFalse(@fopen($path, 'r'), 'precondition');
+
+        $file->setOpenError("fopen() gives error");
+        $this->assertFalse(@fopen($path, 'r'), 'fopen fails now');
+
+        $file->setOpenError(null);
+        $this->assertNotFalse(@fopen($path, 'r'), 'fopen works again');
+    }
 }
```

Comment:
This should be three different tests, so if one fails we know exactly which one. I know a lot of the existing tests don't always adhere to this rule, but new tests should follow the "one assertion per test" rule.